middleware/recordChange.js

```diff
@@ -33,11 +38,19 @@ module.exports = function (req, res, next) {
 				data.theirChange['versionNo'] = theirChange['__v'];
 
 				return;
-			}
+            }
+
+			// if transpose is defined for this field, let's transpose their change so it can compare in the correct format on client side
+			if (getTransposeFn(form, fieldName, 'form')) {
+
+                let transposeFn = getTransposeFn(form, fieldName, 'form')(theirChange[fieldName], theirChange);
+
+                if (!(transposeFn instanceof Promise)) {
```

Comment:
Is this necessary? The Promise support is specific to the contexts `form` and `export` and neither of them have ever supported anything then Promise so I don't think the deprecating is a good thing here? I know the `transpose` itself used to support a function and that we're deprecating that. I don't think the transpose function itself should support Promises. Does this make sense?

Reply:
@smebberson This line has to be wrapped in a promise in order to support the `promises.push` below. `getTransposeFn` is returning the transpose function in the form dsl and this is checking to make sure it is a function (deprecated) or a Promise?